File: src/main/java/social/alone/server/link/Link.java
package social.alone.server.link;

import social.alone.server.event.Event;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import javax.persistence.*;
import java.time.LocalDateTime;

@Getter
@EqualsAndHashCode(of = "id")
@Entity
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Link {


  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  @ManyToOne
  private Event event;

  @Builder
  public Link(Event event) {
    this.event = event;
  }

  @CreationTimestamp
  private LocalDateTime createdAt;

  @UpdateTimestamp
  protected LocalDateTime updatedAt;

}

File: src/main/kotlin/social/alone/server/link/Link.kt
package social.alone.server.link

import social.alone.server.event.domain.Event
import lombok.*
import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.UpdateTimestamp

import javax.persistence.*
import java.time.LocalDateTime

@Getter
@EqualsAndHashCode(of = ["id"])
@Entity
@NoArgsConstructor(access = AccessLevel.PROTECTED)
class Link {


    @Id
    @GeneratedValue
    val id: Long? = null

    @ManyToOne
    val event: Event

    @CreationTimestamp
    val createdAt: LocalDateTime? = null

    @UpdateTimestamp
    var updatedAt: LocalDateTime? = null
        protected set

    @Builder
    constructor(event: Event) {
        this.event = event
    }

}
